languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -2909,8 +2909,28 @@ USA
                 <token spacebefore="no">i</token>
             </antipattern>
             <rule>
+                <!-- antipatterns intended for mathematical or scientific usages of the symbol "i" -->
                 <antipattern>
-                    <token regexp="yes">row|index</token>
+                    <token regexp="yes">(row|index|variable|quantit(y|ie)|number|term|symbol|constant|point|element|unit|value|size|equal)s?|sub|negative|minus|lowercase</token>
+                    <token>i</token>
+                </antipattern>
+                <antipattern>
+                    <token>square</token>
+                    <token>of</token>
+                    <token>i</token>
```

Comment:
I think the AP would be even better without "square". When is the PRP "I" ever preceeded by "of"?
The number i has of before it all the time:

_The distance of i to the origin is 1._